backend/homework/8/src/main/java/com/example/demo/controller/VehicleController.java

```diff
@@ -0,0 +1,67 @@
+package com.example.demo.controller;
+import com.example.demo.dto.VehicleDTO;
+import com.example.demo.model.Vehicle;
+import com.example.demo.services.FactoryService1;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+
+public class VehicleController {
+
+    private FactoryService1 factoryService1;
+    @Autowired  VehicleController(FactoryService1 factoryService1)
+    {
+        this.factoryService1=factoryService1;
+    }
+
+    private List<Vehicle> vehiclesList = new ArrayList<>();
+
+    @PostMapping("factory1/addvehicle")
+    public ResponseEntity<String> addVehicle(@RequestBody @Valid VehicleDTO vehicle)
+    {
+        factoryService1.addVehicle(vehicle);
```

Comment:
Let the service function return ResponseEntity, Incase the function returns an exception the response can still be 200 ok